src/script/view_model/CallingViewModel.ts

```diff
@@ -204,6 +204,10 @@ export class CallingViewModel {
         this.callingRepository.toggleCamera(call);
       },
       toggleMute: (call: Call, muteState: boolean) => {
+        if (call.state() === CALL_STATE.INCOMING) {
+          call.muteState(muteState ? MuteState.SELF_MUTED : MuteState.NOT_MUTED);
```

Comment:
I'd probably move this logic to the callingRepo, I don't understand why the view model would be responsible for this calling logic. WDYT?